app-text/wordgrinder/wordgrinder-0.7.1.recipe

```diff
@@ -48,9 +49,33 @@ BUILD_PREREQUIRES="
 	cmd:pkg_config$secondaryArchSuffix
 	"
 
+if [ -n "$secondaryArchSuffix" ]; then
```

Comment:
Oops, my mistake.
```diff
-if [ -n "$secondaryArchSuffix" ]; then
+if [ -z "$secondaryArchSuffix" ]; then
```
If we don't fix this error then we get no dictionaries sub-package on x86_64, and, on x86_gcc2, `hp wordgrinder_x86` produces an incorrect `wordgrinder_x86_dictionaries-0.7.1-2-x86_gcc2.hpkg` package instead of `wordgrinder_x86_dictionaries-0.7.1-2-any.hpkg` (the difference is mainly on the name: `-any.hpkg` vs `-x86_gcc2.hpkg`)

Yes, haikuporter does not support building arch-independent packages when it is building for a secondary arch.

Reply:
This still doesn't create a dictionaries packages for me?

Reply:
If you are running x86_64 then, with the fixes above, you should obtain a `wordgrinder_x86_dictionaries-0.7.1-2-any.hpkg`.
On x86_gcc2 you won't get that sub-package because wordgrinder does not build on x86_gcc2 primary arch. So in that arch you will get a `wordgrinder_x86-0.7.1-2-x86_gcc2.hpkg`.

Reply:
Still no dictionaries package ...

Reply:
@Begasus, I guess you are running x86_gcc2. It is perfectly normal to not have the dictionaries sub-package when you run "hp wordginder_x86".
That said, you can build it on x86_64 and copy the wordginder_dictionaries hpkg from it to your x86_gcc2 and then install it there (after installing the wordginder_x86 hpkg)

BTW,it could be nice to have a repo for the arch-dependent packages ("-x86_gcc2" and "-x86_64"), and a repo for the arch-independent sub-packages (and packages), ("-any").

Reminder: Arch-independent packages, built on any primary arch, are supposed to be identical.

This way, for packages that do not build on x86_gcc2 primary arch we would be able to install their arch-independent packages from the "-any" repo.

When I say "It would be **nice**", I know this it would **not** be simple.

Reply:
@fbrosson wasn't the plan to make the dictionaries available? Not every one can switch to x86_64 to build the "any" package and then copy them over, IMHO it would be nice if they were able to build on x86_gcc2 also so one wouldn't have to switch?

Reply:
That said, if this is how it should be I'm ok with it

Reply:
Hmm, what we can do is keep the dictionaries in the main package only when we are building for a secondary arch, and, for x86_64, put them in the external package.
This should not be too hard to do. I'll add a review with the changes, then we can decide if that's a good idea or not.